### PR TITLE
US123710 [Engagement] Add Content View metric to Settings Page

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US123710](https://rally1.rallydev.com/#/23525809118d/iterationstatus?detail=%2Fuserstory%2F478982098616&sharedViewId=328666453148&view=dc9e7043-1cf6-486d-a4aa-3a212976e789&fdp=true?fdp=true): [Engagement] Add Content View metric to Settings Page

Related PRs:
https://github.com/Brightspace/lms/pull/7726
https://github.com/Brightspace/valence-api-docs/pull/339

~PR that shows and hides Content View Card will be created after @NicholasHallman merge its PR into master.~

* Functional Testing 
  * [x] E2E test cases
     * `content-view-card` settings is stored and rendered properly for `d2l-insights-engagement-dashboard`
     * `Save and Close` button on Setting Page updates Content View card visability according to `content-view-card` setting
* Security, devops, perf: N/A
* UX and docs
  * [ ] Demo
  
FYI @anhill-D2L @hyehayes @johngwilkinson @NicholasHallman @bemailloux 

image

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
[WIP] Working on hiding the histogram on the main page

File: locales/en.js [hunk 1]
```diff
@@ -187,6 +187,7 @@ export default {
 	"settings:overdueAssignmentsDesc": "The Overdue Assignments card shows the number of users who have one or more assignments overdue in the courses that are filtered on.",
 	"settings:systemAccessDesc": "The System Access card shows the last access the user had in any way.",
 	"settings:discActivityDesc": "The Discussion Activity card shows passive and active social engagement in each course that is filtered on. This metric reports when a user creates a thread, replies to an existing post, or reads a post.",
+	"settings:contentViewDesc": "The Content View card shows how many content topics has been viewed for each enrolment per user.",
```
Comment (bemailloux) [inline locales/en.js:190]: "how many content topics has been viewed" -> "how many content topics **have** been viewed"

Also, I think we usually have been spelling "enrollment" with 2 Ls, although both spellings are technically correct.

**Review by bemailloux — APPROVED** [review]

File: locales/en.js (hunk 1 above)
Comment (Vitalii-Misechko) [reply locales/en.js]: Spelling is fixed. Thanks!

File: test/engagement-dashboard.test.js [hunk 2]
```diff
@@ -105,8 +111,9 @@ describe('d2l-insights-engagement-dashboard', () => {
 					const summaryContainerEl = await trySelect(el.shadowRoot
 						, 'd2l-summary-cards-container');
 
-					allCards.forEach(async card => {
```
Comment (Vitalii-Misechko) [inline test/engagement-dashboard.test.js:108]: forEach + async is not a good idea. `forEach` just does fire and forget. That is why these tests were always green )

File: test/engagement-dashboard.test.js [hunk 3]
```diff
@@ -105,8 +111,9 @@ describe('d2l-insights-engagement-dashboard', () => {
...
-						let renderedCard = await trySelect(el.shadowRoot, `d2l-insights-${card}`, 10);
+					await Promise.all(allCards.map(async card => {
```
Comment (NicholasHallman) [inline test/engagement-dashboard.test.js:114]: 👏

**Review by NicholasHallman — APPROVED** [review]

**Pull request merged**